src/component.cxx

```diff
@@ -24,10 +58,39 @@ bool isSetFinal(const Options& option, const std::string& location) {
   return option.isSet();
 }
 
+bool isSetFinal(const GuardedOptions & option, const std::string& location) {
+  bool set = option.isSet();
+#if CHECKLEVEL >= 1
+  PermissionTypes perm = option.getHighestPermission();
+  if (static_cast<int>(perm) >= static_cast<int>(PermissionTypes::Read)
+      or (perm == PermissionTypes::ReadIfSet and set)) {
+    const Options& opt = option.get();
+    const_cast<Options&>(opt).attributes["final"] = location;
```

Comment:
I really don't like this. It doesn't look like `isSetFinal` is called in any context with a `const&` anyway? I know this is already an issue with the existing function, but can we avoid the `const_cast` here?